File: 768-partition-labels/partition-labels.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        dic = {}
        ans = []
        for i,v in enumerate(s):
            dic[v] = i
        
        end = 0
        size = 0
        for i,v in enumerate(s):
            size += 1
            end = max(end , dic[v])
            if(end == i):
                ans.append(size)
                size = 0
        return ans

File: 768-partition-labels/partition-labels.java
class Solution {
    public List<Integer> partitionLabels(String s) {

        List<Integer> ans = new ArrayList<>();
        Map<Character , Integer> hash = new HashMap<>();

        for(int i=0;i<s.length();i++){
            hash.put(s.charAt(i) , i);
        }

        int end = 0;
        int size = 0;

        for(int i=0;i<s.length();i++){
            size++;

            end = Math.max(end , hash.get(s.charAt(i)));

            if(i == end){
                ans.add(size);
                size = 0;
            }
        }
    return ans;

    }
}